### PR TITLE
Upgrade future and clean up instruction result object

### PR DESCRIPTION
**Pull request by Toxantron**

### CODE REVIEW

File: src/Moryx.ControlSystem/VisualInstructions/ActiveInstruction.cs [hunk 1]
```diff
@@ -31,8 +32,8 @@ public class ActiveInstruction
         public object Inputs { get; set; }
 
         /// <summary>
-        /// Results of the instruction
+        /// Possible results of the instruction
         /// </summary>
-        public IReadOnlyList<string> PossibleResults { get; set; } = new string[0];
+        public IReadOnlyList<InstructionResult> Results { get; set; } = Array.Empty<InstructionResult>();
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/ActiveInstruction.cs:37]: For the `EnumInstructionResult` you left the old method `PossibleInstructionResults` in there with an obsolete attribute. Here I would suggest either doing it the same way, or specifically adding the Renaming of this property to the migration guide. otherwise people will just wonder why the property isn't there anymore and hope, that the new property does the same thing after they found it.

File: src/Moryx.ControlSystem/VisualInstructions/ActiveInstructionResponse.cs [hunk 2]
```diff
@@ -26,6 +25,6 @@ public class ActiveInstructionResponse
         /// <summary>
         /// Selected result option by the user
         /// </summary>
-        public string Result { get; set; }
+        public InstructionResult SelectedResult { get; set; }
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/ActiveInstructionResponse.cs:28]: Same as above

File: src/Moryx.ControlSystem/VisualInstructions/VisualInstructorExtensions.cs [hunk 3]
```diff
@@ -77,20 +77,20 @@ public static long Execute(this IVisualInstructor instructor, string title, IVis
                 title,
                 parameter,
                 EnumInstructionResult.PossibleResults(typeof(T)),
-                result => { callback(EnumInstructionResult.ResultToEnumValue(typeof(T), result.Result)); });
+                result => callback(EnumInstructionResult.ResultToGenericEnumValue<T>(result.SelectedResult)));
```
Comment (1nf0rmagician) [inline src/Moryx.ControlSystem/VisualInstructions/VisualInstructorExtensions.cs:80]: Maybe this is also an example to put into the migration guide. At least I couldn't make out on first glance why this change had to be done in the way you did it 🤔

File: src/Moryx.Orders/Operation.cs [hunk 4]
```diff
@@ -108,13 +108,23 @@ public class Operation
...
+        }
 
         /// <summary>
         /// Current state classification of this operation
         /// </summary>
         public virtual OperationClassification FullState { get; protected set; }
 
+        /// <summary>
+        /// Detailed display name of the state
+        /// TODO: Remove this property in next major and replace rework OperationClassification
```
Comment (1nf0rmagician) [inline src/Moryx.Orders/Operation.cs:124]: Here is an open Todo which has nothing to do with the change from @dbeuchler regarding the notification from OrderData

File: src/Moryx.ControlSystem/VisualInstructions/ActiveInstruction.cs (hunk 1 above)
Comment (Toxantron) [reply src/Moryx.ControlSystem/VisualInstructions/ActiveInstruction.cs:37]: That is technically not a renaming. It removes the obsolote from `dev`, Git just shows the diff weird. In `EnumInstructionResult` I also removed the string version and renamed the method introduced with 8.1, but kept the old-new method.

File: src/Moryx.ControlSystem/VisualInstructions/ActiveInstructionResponse.cs (hunk 2 above)
Comment (Toxantron) [reply src/Moryx.ControlSystem/VisualInstructions/ActiveInstructionResponse.cs:28]: Dito :-P

File: src/Moryx.ControlSystem/VisualInstructions/VisualInstructorExtensions.cs (hunk 3 above)
Comment (Toxantron) [reply src/Moryx.ControlSystem/VisualInstructions/VisualInstructorExtensions.cs:80]: Because it did not compile, the old method still exists, but here an overload was called that did not match the caller signature.

File: src/Moryx.Orders/Operation.cs (hunk 4 above)
Comment (Toxantron) [reply src/Moryx.Orders/Operation.cs:124]: Don't know why this was a TODO, DisplayName is a great idea regardless of state refactoring
